src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicImplementationProvider.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Net.Quic.Implementations.MsQuic.Internal;
+using System.Net.Security;
+
+namespace System.Net.Quic.Implementations.MsQuic
+{
+    internal sealed class MsQuicImplementationProvider : QuicImplementationProvider
+    {
+        // TODO these "session" objects should probably be statics and only create one.
+        // No need to create two different sessions for client and server.
+        private MsQuicSession _clientSession = new MsQuicSession("client");
+        private MsQuicSession _serverSession = new MsQuicSession("server");
```

Comment:
This way of dealing with sessions won't work correctly with 0-RTT/resumption. The app layer is going to need to take ownership of these most likely. Otherwise you need to keep a table of sessions, and if the app creates a new connection with the same ALPN, use the same previous session (unless they don't want to resume...?).

Reply:
Going to comment on this in another response. Trying to figure out the right thing to do here.